kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/activity/MobiComAttachmentSelectorActivity.java

```diff
@@ -260,71 +223,26 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
-            if (selectedFileUri != null) {
-                String fileName = null;
-                try {
-                    long maxFileSize = alCustomizationSettings.getMaxAttachmentSizeAllowed() * 1024 * 1024;
-                    Cursor returnCursor =
-                            getContentResolver().query(selectedFileUri, null, null, null, null);
-                    if (returnCursor != null) {
-                        int sizeIndex = returnCursor.getColumnIndex(OpenableColumns.SIZE);
-                        returnCursor.moveToFirst();
-                        Long fileSize = returnCursor.getLong(sizeIndex);
-                        returnCursor.close();
-                        if (fileSize > maxFileSize) {
-                            Toast.makeText(this, R.string.info_attachment_max_allowed_file_size, Toast.LENGTH_LONG).show();
-                            return;
-                        }
-                    }
-                    String mimeType = FileUtils.getMimeTypeByContentUriOrOther(this, selectedFileUri);
-                    if (TextUtils.isEmpty(mimeType)) {
-                        return;
-                    }
-                    if (!checkMimeType(mimeType)) {
-                        Toast.makeText(this, R.string.info_file_attachment_mime_type_not_supported, Toast.LENGTH_LONG).show();
-                        return;
-                    }
-                    String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
-                    fileName = FileUtils.getFileName(this, selectedFileUri);
+            int returnCode = kmAttachmentsController.processFile(selectedFileUri, alCustomizationSettings, prePostUIMethodsFileAsyncTask);
 
-                    String fileFormat = FileUtils.getFileFormat(fileName);
-                    String fileNameToWrite;
-                    if (TextUtils.isEmpty(fileFormat)) {
-                        String format = FileUtils.getFileFormat(FileUtils.getFile(this, selectedFileUri).getAbsolutePath());
-                        if (TextUtils.isEmpty(format)) {
-                            return;
-                        }
-                        fileNameToWrite = timeStamp + "." + format;
-                    } else {
-                        fileNameToWrite = timeStamp + "." + fileFormat;
-                    }
-                    File mediaFile = FileClientService.getFilePath(fileNameToWrite, getApplicationContext(), mimeType);
-                    new FileTaskAsync(mediaFile, selectedFileUri, this).execute((Void) null);
-                } catch (Exception e) {
-                    e.printStackTrace();
-                }
+            switch (returnCode) {
+                case -1:
+                    Toast.makeText(this, R.string.info_attachment_max_allowed_file_size, Toast.LENGTH_LONG).show();
+                    break;
+                case -2:
+                    Utils.printLog(this, TAG, "URI mime type is empty.");
+                    break;
```

Comment:
Why are 2 as toast and 2 as log?

Reply:
the two that are logs have information that the end users doesn't have to know. the mime type will never be empty unless there is some issue in the code. and most probably before this, -1 will be returned.